lib/providers/token.js

```diff
@@ -202,6 +202,10 @@ class TokenProvider {
    * @private
    */
   _post(payload) {
+    const payloadStr = JSON.stringify(payload);
+
+    this._warden.headers['Content-Length'] = payloadStr.length;
```

Comment:
@dgreene-r7 pointed out that `_post` is only called one (as a result of `initialize()`) so we won't run into the case where we have two different payloads within the same instance.